Assets/Scripts/PlayerController.cs

```diff
@@ -551,6 +576,35 @@ private bool CanUseButton()
     {
         return !_buttonUsed && _time <= _timeButtonPressed + buttonBufferTime;
     }
+  
+    
+
+    private bool EpsilonGround()
+    {
+        float dist = DistanceToGround(); 
+        if ( dist < 0.5f)
+        {
+            return true;
+        }
+
+        return false; 
+    }
+    
+    private float DistanceToGround()
+    {
+        Vector2 direction = Vector2.down;
+        float rayDistance = 10f; // Maximum distance to check
```

Comment:
to make this more useful in general, will be better to just always return the distance to ground and not have a edge case where we return -1